dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/IAdrServiceClient.cs

```diff
@@ -0,0 +1,128 @@
+    /// <param name="aepName">The name of the Asset Endpoint Profile.</param>
+    /// <param name="request">The request containing asset retrieval details.</param>
+    /// <param name="commandTimeout">Optional timeout for the command execution.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>The requested asset details.</returns>
+    Task<Asset> GetAssetAsync(string aepName,
+        GetAssetRequest request,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Updates the status of a specified asset within an Asset Endpoint Profile (AEP).
+    /// </summary>
+    /// <param name="aepName">The name of the Asset Endpoint Profile.</param>
+    /// <param name="request">The request containing the asset status update details.</param>
+    /// <param name="commandTimeout">Optional timeout for the command execution.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>The updated asset details.</returns>
+    Task<Asset> UpdateAssetStatusAsync(string aepName,
+        UpdateAssetStatusRequest request,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Creates a detected asset within an Asset Endpoint Profile (AEP).
+    /// </summary>
+    /// <param name="aepName">The name of the Asset Endpoint Profile.</param>
+    /// <param name="request">The request containing details of the detected asset.</param>
+    /// <param name="commandTimeout">Optional timeout for the command execution.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>The response containing details of the created detected asset.</returns>
+    Task<CreateDetectedAssetResponse> CreateDetectedAssetAsync(string aepName,
+        CreateDetectedAssetRequest request,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Creates a discovered Asset Endpoint Profile (AEP).
+    /// </summary>
+    /// <param name="aepName">The name of the Asset Endpoint Profile.</param>
+    /// <param name="request">The request containing details of the discovered AEP.</param>
+    /// <param name="commandTimeout">Optional timeout for the command execution.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>The response containing details of the created discovered AEP.</returns>
+    Task<CreateDiscoveredAssetEndpointProfileResponse> CreateDiscoveredAssetEndpointProfileAsync(string aepName,
+        CreateDiscoveredAssetEndpointProfileRequest request,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Event triggered when telemetry updates for an Asset Endpoint Profile (AEP) are received.
```

Comment:
I'd call out in these headers that users must turn on telemetry updates by calling the Observe APIs here. Otherwise a user may think they just need to set this callback